hazelcast/src/main/java/com/hazelcast/sql/impl/schema/map/AbstractMapTable.java

```diff
@@ -37,6 +37,7 @@
     private final UpsertTargetDescriptor keyUpsertDescriptor;
     private final UpsertTargetDescriptor valueUpsertDescriptor;
     private final QueryException exception;
+    private final boolean nativeMemoryEnabled;
```

Comment:
I think this flag should be in the `PartitionedMapTable`, because replicated maps do not support indexes, and hence can never be used with HD. Old logic already handles this properly.